src/main/java/org/spongepowered/api/data/type/RailDirection.java

```diff
@@ -24,11 +24,103 @@
  */
 package org.spongepowered.api.data.type;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
 import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.Sponge;
 import org.spongepowered.api.util.Cycleable;
+import org.spongepowered.api.util.Direction;
 import org.spongepowered.api.util.annotation.CatalogedBy;
 
+import java.util.Optional;
+
 @CatalogedBy(RailDirections.class)
 public interface RailDirection extends CatalogType, Cycleable<RailDirection> {
 
+    /**
+     * Returns the {@link RailDirection} from the given facing directions.
+     *
+     * @param firstDirection The first direction
+     * @param secondDirection The second direction
+     * @return The {@link RailDirection}
+     * @throws IllegalArgumentException If the given directions don't differ
+     */
+    static RailDirection fromFaces(final Direction firstDirection, final Direction secondDirection) {
```

Comment:
I do not like these static methods having impl code. Can we create a builder and have these methods call this builder?